scipy/sparse/dok.py

```diff
@@ -93,28 +95,41 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False):
             if dtype is not None:
                 arg1 = arg1.astype(dtype)
 
-            self.update(arg1)
+            dict.update(self, arg1)
             self.shape = arg1.shape
             self.dtype = arg1.dtype
         else:  # Dense ctor
             try:
                 arg1 = np.asarray(arg1)
             except:
-                raise TypeError('invalid input format')
+                raise TypeError('Invalid input format.')
 
             if len(arg1.shape) != 2:
-                raise TypeError('expected rank <=2 dense array or matrix')
+                raise TypeError('Expected rank <=2 dense array or matrix.')
 
             from .coo import coo_matrix
             d = coo_matrix(arg1, dtype=dtype).todok()
-            self.update(d)
+            dict.update(self, d)
             self.shape = arg1.shape
             self.dtype = d.dtype
 
+    def update(self, val):
+        # Prevent direct usage of update
+        raise NotImplementedError("Direct modification to dok_matrix element "
+                                  "is not allowed.")
+
+    def _update(self, data):
+        """ An update method for dict data defined for direct access to
+        `dok_matrix` data. Main purpose is to be used for effient conversion
+        from other spmatrix classes. Has no checking if `data` is valid.
+        """
+        return dict.update(self, data)
+
+
     def getnnz(self, axis=None):
         if axis is not None:
-            raise NotImplementedError("getnnz over an axis is not implemented "
-                                      "for DOK format")
+            raise NotImplementedError("Getnnz over an axis is not implemented "
```

Comment:
Don't capitalize the method name. Could reorder like: "DOK format doesn't support getnnz() over an axis"